ludwig/data/preprocessing.py

```diff
@@ -1028,7 +1036,29 @@ def build_dataset(
     return dataset, metadata
 
 
-def build_metadata(dataset_df, features, global_preprocessing_parameters, backend):
+def cast_columns(input_df, features, global_preprocessing_parameters, backend):
+    # todo figure out if global_preprocessing_parameters is needed
+    proc_df = backend.df_engine.empty_df_like(input_df)
```

Comment:
Similar to the way we do missing value replacement, I think we should apply these cases directly to the input_df. Otherwise, the `proc_df` will contain both the input columns like `text` as well as the processed columns like `text_HDSDP1393`.  We only want processed columns in the `proc_df`.